packages/xod-tethering-inet/src/Connection.re

```diff
@@ -0,0 +1,99 @@
+open Belt;
+
+module Host = {
+  type t = string;
+
+  let ipRegExp = [%re {|/^(\d{1,3}).(\d{1,3}).(\d{1,3}).(\d{1,3})$/|}];
+  let hostnameRegExp = [%re
+    {|/^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$/|}
+  ];
+
+  let isIp = Js.Re.test_(ipRegExp);
+  let isHostname = Js.Re.test_(hostnameRegExp);
+
+  let validate = str =>
+    isIp(str) || isHostname(str)
+      ? Result.Ok(str) : Result.Error(Error.INVALID_HOSTNAME(str));
+};
+module Port = {
+  type t = int;
+
+  let fromString = str =>
+    Int.fromString(str)
+    |> (
+      res =>
+        switch (res) {
+        | Some(a) when a > 0 => Result.Ok(a)
+        | _ => Result.Error(Error.INVALID_PORT(str))
+        }
+    );
```

Comment:
This might be simplified to:

```reason
let fromString = str =>
  switch(Int.fromString(str)) {
  | Some(a) when a > 0 => Result.Ok(a)
  | _ => Result.Error(Error.INVALID_PORT(str))
  }
```

Not a big deal, though